tests_e2e/pipeline/scripts/execute_tests.sh

```diff
@@ -2,36 +2,62 @@
 
 set -euxo pipefail
 
+#
+# Set the correct mode for the private SSH key and generate the public key.
+#
+cd "$HOME"
+mkdir ssh
+cp "$DOWNLOADSSHKEY_SECUREFILEPATH" ssh
+chmod 700 ssh/id_rsa
+ssh-keygen -y -f ssh/id_rsa > ssh/id_rsa.pub
```

Comment:
this piece of code to setup the SSH key comes from the previous run-scenarios script, which I am deleting in this PR